src/darwin/Framework/CHIP/MTROTAImageTransferHandler.mm

```diff
@@ -272,6 +286,29 @@ - (instancetype)initWithMTROTAImageTransferHandler:(MTROTAImageTransferHandler *
                           }];
     };
 
+    __block void (^completionHandler)(NSData * _Nullable data, BOOL isEOF) = nil;
+
+    // If the peer node is a Thread device, check how much time has elapsed since we started processing the BlockQuery.
+    // If the time elapsed is greater than kBdxThrottleIntervalInMsecs, call the completion handler to respond with a Block right away.
```

Comment:
This does not match the old polling behavior or the behavior we discussed on the call...  But maybe it's fine.

The difference is that if 51ms have elapsed the old behavior would wait another 49ms, while the new one is not waiting at all.

Reply:
yeah it always waits 50 msecs which should be ok. and we are making this configurable so we wait x amount of time that's can be set at runtime so maybe its fine

Reply:
added a configurable timer